lib/src/bazelify/build.dart

```diff
@@ -485,6 +502,88 @@ class DdcDevServer implements DartBuildRule {
   }
 }
 
+/// A `dart_builder_binary` definition.
+class DartBuilderBinary implements DartBuildRule {
+  @override
+  Iterable<String> get dependencies => [target];
+
+  @override
+  Iterable<String> get excludeSources => null;
+
+  @override
+  final String name;
+
+  @override
+  final String package;
+
+  @override
+  Iterable<String> get sources => null;
+
+  /// The name of the custom `Builder` class.
+  final String clazz;
+
+  /// The name of the constructor to use, if `null` then the default constructor
+  /// will be used.
+  final String constructor;
+
+  /// The import to be used to load `clazz`.
+  final String import;
+
+  /// The input extension to treat as primary inputs to the builder.
+  final String inputExtension;
+
+  /// The expected output extensions.\
+  ///
+  /// For each file matching `inputExtension` a matching file with each of
+  /// these extensions must be output.
```

Comment:
Well the action must output the files, so as far as the build rule is concerned yes its necessary.